// @connections/rest.mjs
/**
 * Rest helpers to connect with json api rest endpoints.
 */

// import fetch from 'node-fetch'
import axios from 'axios'


function send({ method, path, data, token }) {
	// const fetch = process.browser ? window.fetch : require('node-fetch').default;

	const opts = { method, headers: {} , data};

	if (data) {
		opts.headers['Content-Type'] = 'application/json';
	}

	if (token) {
		opts.headers['Authorization'] = `Token ${token}`;
	}

  if (path) {
    return axios(`${path}`, opts)
		.then(json => {
			try {
        console.log('person created ', json)
				return json
			} catch (err) {
				return null;
			}
		}).catch(err => {console.log(err); return err;} );
  } else {
    return null
  }


}

function get(path, token) {
  const data = null
	return send({ method: 'GET', path, data, token });
}

function del(path, token) {
  const data = null

	return send({ method: 'DELETE', path, data, token });
}

function post(path, data, token) {
  // return axios.post(path, data).then(res =>  res).catch( err => console.log(err))
  return send({method: 'POST', path, data, token })
}

function put(path, data, token) {
	return send({ method: 'PUT', path, data, token });
}

function patch(path, data, token) {
	return send({ method: 'PATCH', path, data, token });
}

export { get , del , post, put , patch }
